test/viewmodels/challenge_view_model_integration_test.dart

```diff
@@ -193,6 +183,47 @@ void main() {
       final points = updatedUser.data.centauriPoints;
       expect(points, ChallengeRepositoryService.soloChallengeReward);
     });
+
+    test("Challenges position is updated on user position change and refresh",
+        () async {
+      final posts = postGenerator.generatePostsAt(userPosition1, 3);
+      setPostsFirestore(posts, fakeFireStore);
+
+      final activeChallenges =
+          challengeGenerator.generateChallenges(1, false, extraTime);
+
+      await setChallenges(
+        fakeFireStore,
+        activeChallenges,
+        testingUserFirestoreId,
+      );
+
+      var challenges = await container.read(challengeViewModelProvider.future);
```

Comment:
I wonder if we could have 2 `final` variables here instead of a single `var`.
Maybe something like `challengesBeforeRefresh` and `challengesAfterRefresh`.
I think it could be more explicit :)

We could have something like that I believe :

```dart
      final challengesBeforeRefresh =
          await container.read(challengeViewModelProvider.future);

      // Check initial distances are 0 since user is at userPosition1
      for (var challenge in challengesBeforeRefresh) {
        expect(challenge.distance, 0);
      }

      // Change user position to userPosition2
      when(geoLocationService.getCurrentPosition()).thenAnswer(
        (_) async => userPosition2,
      );

      // Refresh challenges
      await container.read(challengeViewModelProvider.notifier).refresh();
      final challengesAfterRefresh =
          await container.read(challengeViewModelProvider.future);

      // Compute the distance between userPosition1 and userPosition2
      final double distanceKm = const GeoFirePoint(userPosition1)
          .distanceBetweenInKm(geopoint: userPosition2);

      final int distanceM = (distanceKm * 1000).toInt();

      // Check that challenges distances are updated correctly
      for (var challenge in challengesAfterRefresh) {
        expect(challenge.distance, distanceM);
      }
```

Reply:
Fixed in 66ff5e1c962070c580d9b3943e0a9e6d5a61cb11